### PR TITLE
object-manipulation-2

### PR DESCRIPTION
**Pull request by matt-cha**
Screen Shot 2022-12-27 at 8 21 34 PM

### CODE REVIEW

File: object-manipulation-2/src/invert.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+hold the number of key value pairs in the source
+-add the key value pair to the new object created by assigning the
+values variable at each index of i as the key,
+-equal to the keys variable at index of i as the value  */
+function invert(source) {
+  var newObject = {}; //
+  var newValues = [];
+  var newKeys = [];
+  newKeys = Object.keys(source);
+  newValues = Object.values(source);
```
Comment (codingcodymiller) [inline object-manipulation-2/src/invert.js:17]: Try to do this without turning the object into an array 🤓

File: object-manipulation-2/src/omit.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* exported omit */
+/* -create a function omit with 2 arguments
+-create an empty new object
+-for loop that checks through the keys length
+-set key equal to the keys at object source at index of i
+-if key is equal to undefined,
+-set the new object's keys at index i value to the new key variable */
+function omit(source, keys) {
+  var newObject = {};
+  var sourceKeys = Object.keys(source);
```
Comment (codingcodymiller) [inline object-manipulation-2/src/omit.js:10]: Try to do this without turning the object into an array 🤓

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: array-manipulation-2/src/chunk.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* exported chunk */
+/* -create a function with 2 arguments
+-create a empty array that will contain the array of arrays
+-need to create one main array that returns  */
+/* function chunk(array, size) {
+  var newArray = array.split('', 0, size);
```
Comment (RobertGardner) [inline array-manipulation-2/src/chunk.js:6]: `split` is a method on String, not on Array. I'm not sure what you are trying to do here, so I can't help!

File: array-manipulation-2/src/chunk.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  console.log(newArray);
+  for (var i = 0; i < size; i++) {
+    newArray.push(array[i]);
+  }
+  return newArray.join('');
+} */
+/* need to use slice? */
```
Comment (RobertGardner) [inline array-manipulation-2/src/chunk.js:13]: This code using slice is a good start. You probably need to loop, slicing out a chunk array each time.

Another way to approach this (which doesn't use slice) is to loop through the array and push each element into a `chunk` array. After pushing the `size`th element into `chunk`, push that chunk into the `chunks` array (`newArray` in your code) and start a new `chunk` array.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Not a bad start, but more work is needed on some of the exercises.

File: array-manipulation-2/src/chunk.js (hunk 4 above)
Comment (matt-cha) [reply array-manipulation-2/src/chunk.js:13]: This is a bit of a mess but these fixes are for array-manipulation-2 and this exercise should be object-manipulation-2! 

I accidentally saved these changes when doing the exercises simultaneously on the wrong branch so I had chunk.js finished under array-manipulation-2 exercise. But I did fix the other problems above that you commented on + copied my code for chunk.js. 

Sorry for the confusion Robert! If you want to check again for these commented solutions even though it is for a different exercise that'd be great thanks!

**Pull request merged**